demo/client/backend/src/fabricClient.js

```diff
@@ -0,0 +1,477 @@
+                 console.log('Error in register(): ' + error);
+                 console.log("\n----------------------------------------");
+                 return error;
+             });
+}  //  end of function registerUser
+
+async function enrollUser (userName, secret) {
+    console.log (">>> Entry: function, enrollUser...");
+    try {
+        // Check to see if we have already enrolled the user.
+        const userExists = await wallet.exists(userName);
+        if (userExists) {
+           var result =  {
+                'status': {
+                  'rc': USER_EXISTS,
+                  'msg': 'An identity for the user ' + userName + ' already exists in the wallet' }
+              }
+
+        console.log('An identity for the user ' + userName + ' already exists in the wallet');
+        return result;
+        }
+
+        //  get CA url from connection_profile
+        // Create a new CA client for interacting with the CA.
+        const caURL = connectionProfile.certificateAuthorities['ca.example.com'].url;
+        const caclient = new FabricCAClient(caURL);
+
+        const enrollment = await caclient.enroll(
+            { enrollmentID: userName,
+              enrollmentSecret: secret});
+        //    affiliation: org1Affiliation });
+
+        const identity = await X509WalletMixin.createIdentity(orgName, enrollment.certificate, enrollment.key.toBytes());
+        await wallet.import(userName, identity);
+
+        //  return success
+        var result =
+        {
+          "status": {  'rc': SUCCESS,
+          'msg': 'Successfully enrolled user ' + userName + 'and imported it into the wallet' }
+        };
+        console.log('Successfully enrolled user ' + userName + ' and imported it into the wallet');
+        return result;
+
+    } catch (error) {
+      var result =
+      {
+        "status": {  'rc': USER_UNAUTHORIZED,
+        'msg': 'Failed to enroll user ' + userName + ': ${error}'  }
+      };
+
+      console.error( 'Failed to enroll user ' + userName + ': ${error}');
+      throw (result);
+    }
+
+    //  console.log ("<<< Exit: function, enrollUser");
+}
+
+
+//  function setUserContext
+//  Purpose:  to set the context to the user (who called this api)
+//            All subsequent calls using that gateway / contract will be on this user's behalf.
+//  Input:  userName - which has been registered;  If not enrolled, this function enrolls.
+//  Output:  if not error returns SUCCESS else throws an exception
+//          Also, (Global variable) contract will be set to this user's context
+//          All further transactions using this contract will be submitted using
+//          this userName
+async function setUserContext (userName, pwd) {
+    console.log ('>>> Entry: function setUserContext ....');
+
+    try {
+        //  verify if user is registered
+        await enrollUser(userName, pwd);
+    } catch (error) {
+      throw (error);
+    }
```

Comment:
when reading about async functions and await (https://javascript.info/async-await) seems that you can just call await encrollUser(...) and if this throws an error ... you can catch it later with setUserContext.catch( err => ....

Reply:
Again, this was dev code without _any_  scrutiny done, trying to bring in a basic dev code for end-to-end integration.  

Specifically, this try-catch block was intended (as a reminder) to catch errors from enrollUser so that they can be handled differently.    In later commit, the error is converted to a status object.